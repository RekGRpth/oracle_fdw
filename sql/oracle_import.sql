SET client_min_messages = WARNING;

CREATE SCHEMA import;

/* first, import only the table */
IMPORT FOREIGN SCHEMA "SCOTT"
   LIMIT TO ("typetest1", ttv)
   FROM SERVER oracle
   INTO import
   OPTIONS (case 'lower', collation 'C', skip_views 'true');

SELECT t.relname, fs.srvname, ft.ftoptions
FROM pg_foreign_table ft
     JOIN pg_class t ON ft.ftrelid = t.oid
     JOIN pg_foreign_server fs ON ft.ftserver = fs.oid
WHERE relnamespace = 'import'::regnamespace;

/* then import only the view */
IMPORT FOREIGN SCHEMA "SCOTT"
   LIMIT TO ("typetest1", ttv)
   FROM SERVER oracle
   INTO import
   OPTIONS (case 'lower', collation 'C', skip_tables 'true');

SELECT t.relname, fs.srvname, ft.ftoptions
FROM pg_foreign_table ft
     JOIN pg_class t ON ft.ftrelid = t.oid
     JOIN pg_foreign_server fs ON ft.ftserver = fs.oid
WHERE relnamespace = 'import'::regnamespace;

SELECT attname, atttypid::regtype, attfdwoptions
FROM pg_attribute
WHERE attrelid = 'typetest1'::regclass
  AND attnum > 0
  AND NOT attisdropped
ORDER BY attnum;
