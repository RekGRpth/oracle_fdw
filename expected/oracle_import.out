SET client_min_messages = WARNING;
CREATE SCHEMA import;
/* first, import only the table */
IMPORT FOREIGN SCHEMA "SCOTT"
   LIMIT TO ("typetest1", ttv)
   FROM SERVER oracle
   INTO import
   OPTIONS (case 'lower', collation 'C', skip_views 'true');
SELECT t.relname, fs.srvname, ft.ftoptions
FROM pg_foreign_table ft
     JOIN pg_class t ON ft.ftrelid = t.oid
     JOIN pg_foreign_server fs ON ft.ftserver = fs.oid
WHERE relnamespace = 'import'::regnamespace;
  relname  | srvname |           ftoptions            
-----------+---------+--------------------------------
 typetest1 | oracle  | {schema=SCOTT,table=TYPETEST1}
(1 row)

/* then import only the view */
IMPORT FOREIGN SCHEMA "SCOTT"
   LIMIT TO ("typetest1", ttv)
   FROM SERVER oracle
   INTO import
   OPTIONS (case 'lower', collation 'C', skip_tables 'true');
SELECT t.relname, fs.srvname, ft.ftoptions
FROM pg_foreign_table ft
     JOIN pg_class t ON ft.ftrelid = t.oid
     JOIN pg_foreign_server fs ON ft.ftserver = fs.oid
WHERE relnamespace = 'import'::regnamespace;
  relname  | srvname |           ftoptions            
-----------+---------+--------------------------------
 typetest1 | oracle  | {schema=SCOTT,table=TYPETEST1}
 ttv       | oracle  | {schema=SCOTT,table=TTV}
(2 rows)

SELECT attname, atttypid::regtype, attfdwoptions
FROM pg_attribute
WHERE attrelid = 'typetest1'::regclass
  AND attnum > 0
  AND NOT attisdropped
ORDER BY attnum;
 attname |         atttypid         | attfdwoptions 
---------+--------------------------+---------------
 id      | integer                  | {key=yes}
 c       | character                | 
 nc      | character                | 
 vc      | character varying        | 
 nvc     | character varying        | 
 lc      | text                     | 
 r       | bytea                    | 
 u       | uuid                     | 
 lb      | bytea                    | 
 lr      | bytea                    | 
 b       | boolean                  | 
 num     | numeric                  | 
 fl      | double precision         | 
 db      | double precision         | 
 d       | date                     | 
 ts      | timestamp with time zone | 
 ids     | interval                 | 
 iym     | interval                 | 
(18 rows)

